lib/src/resources/movies_api_provider.dart

```diff
@@ -1,66 +1,78 @@
 import 'dart:async';
 import 'dart:convert';
 import 'package:http/http.dart' as http;
-import '../utils/movie_string.dart';
 import '../model/movies.dart';
+import '../utils/movies_string.dart';
 
 class MoviesApiProvider {
   http.Client client = http.Client();
 
   Future<Movies> fetchTrendingMovies() async {
-    final trendingResponse = await client.get(
-      Uri.parse(
-        MovieStrings.trendingMoviesEndpoint + MovieStrings.apiKey,
-      ),
-    );
-    if (trendingResponse.statusCode == 200) {
-      return Movies.fromJson(
-        json.decode(
-          trendingResponse.body,
+    try {
+      final trendingResponse = await client.get(
+        Uri.parse(
+          MoviesStrings.trendingMoviesEndpoint + MoviesStrings.apiKey,
         ),
       );
-    } else {
-      throw Exception(
-        MovieStrings.moviesLoadingError,
-      );
+      if (trendingResponse.statusCode == 200) {
+        return Movies.fromJson(
+          json.decode(
+            trendingResponse.body,
+          ),
+        );
+      } else {
+        return Movies(
+            error: true, page: 0, totalPages: 0, totalResults: 0, results: []);
+      }
+    } catch (e) {
+      return Movies(
+          error: true, page: 0, totalPages: 0, totalResults: 0, results: []);
     }
   }
 
   Future<Movies> fetchDiscoverMovies() async {
-    final discoverResponse = await client.get(
-      Uri.parse(
-        MovieStrings.discoverMoviesEndpoint,
-      ),
-    );
-    if (discoverResponse.statusCode == 200) {
-      return Movies.fromJson(
-        json.decode(
-          discoverResponse.body,
+    try {
+      final discoverResponse = await client.get(
+        Uri.parse(
+          MoviesStrings.discoverMoviesEndpoint,
         ),
       );
-    } else {
-      throw Exception(
-        MovieStrings.moviesLoadingError,
-      );
+      if (discoverResponse.statusCode == 200) {
+        return Movies.fromJson(
+          json.decode(
+            discoverResponse.body,
+          ),
+        );
+      } else {
+        return Movies(
+            error: true, page: 0, totalPages: 0, totalResults: 0, results: []);
```

Comment:
In what cases this can be an error and when this is not an error?

Reply:
done